base/threadingconstructs.jl

```diff
@@ -65,15 +62,12 @@ function _threadsfor(iter,lbody)
             end
         end
         end
-        # Hack to make nested threaded loops kinda work
-        if threadid() != 1 || in_threaded_loop[]
-            # We are in a nested threaded loop
+        if threadid() != 1
+            # only thread 1 can enter/exit _threadedregion
             Base.invokelatest(threadsfor_fun, true)
         else
-            in_threaded_loop[] = true
             # the ccall is not expected to throw
```

Comment:
This comment should also be removed. It's exactly what justified the code here with the previous implementation.